devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy

```diff
@@ -0,0 +1,334 @@
+			state "battery", label: 'Battery:\n${currentValue}%', unit: "%"
+		}
+
+		standardTile("refresh", "command.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "refresh", label: 'refresh', action: "refresh.refresh", icon: "st.secondary.refresh-icon"
+		}
+		main "thermostat"
+		details(["thermostat", "battery", "refresh"])
+	}
+}
+
+def installed() {
+	runIn(2, "initialize")
+}
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	def supportedModes = ["off", "emergency heat", "heat"]
+	state.supportedModes = supportedModes
+	sendEvent(name: "temperature", value: 0)
+	sendEvent(name: "supportedThermostatModes", value: supportedModes, displayed: false)
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	response(refresh())
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	} else {
+		log.warn "${device.displayName} - no-parsed event: ${description}"
+	}
+	log.debug "Parse returned: ${result}"
+	return result
+}
+
+private commandClasses() {
+	[0x70: 1]
+}
+
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand(commandClasses())
+	if (encapsulatedCommand) {
+		log.debug "SecurityMessageEncapsulation into: ${encapsulatedCommand}"
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "unable to extract secure command from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand(commandClasses())
+	if (encapsulatedCommand) {
+		log.debug "MultiChannel Encapsulation: ${encapsulatedCommand}"
+		zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint)
+	} else {
+		log.warn "unable to extract multi channel command from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd, sourceEndPoint = null) {
+	def value = cmd.batteryLevel == 255 ? 1 : cmd.batteryLevel
+	def map = [name: "battery", value: value]
+	switch(sourceEndPoint) {
+		case 1:
+			createEvent(map)
+			break
+		case 2:
+			if(childDevices) {
+				sendEventToChild(map)
+			}
+			break
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatmodev2.ThermostatModeReport cmd, sourceEndPoint = null) {
+	def mode
+	switch (cmd.mode) {
+		case 1:
+			mode = "heat"
+			break
+		case 31:
+			mode = "emergency heat"
+			break
+		case 0:
+			mode = "off"
+			break
+	}
+
+	createEvent(name: "thermostatMode", value: mode, data: [supportedThermostatModes: state.supportedModes])
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointReport cmd, sourceEndPoint = null) {
```

Comment:
I'd remove the `sourceEndPoint` arguments for `zwaveEvent`s where it's not possible to have one